crates/nu-command/src/filesystem/ls.rs

```diff
@@ -969,8 +976,14 @@ fn read_dir(
                 .map_err(ShellError::from)
         });
     if !use_threads {
-        let mut collected = items.collect::<Vec<_>>();
+        let mut collected = items
+            .map(|i| {
+                signals.check(span)?;
+                Ok(i)
+            })
```

Comment:
While I guess this protects from stalling in the `FromIterator` stage, this only seems to introduce a checkpoint for the `!use_threads` case. Are we guarding the `use_threads` case in another way?

Would it make sense to lift this into the upper `map` closer to the I/O operations which may take even longer compared to the alloc?

Reply:
OK if the other branch is interrupting its fine to not duplicate that check. But whatever is readable enough so it doesn't break easily on the next refactor/feature addition is fine by me.